lib/wasix/src/runtime/task_manager/mod.rs

```diff
@@ -91,6 +97,7 @@ pub struct TaskWasm<'a> {
     pub spawn_type: SpawnMemoryType<'a>,
     pub trigger: Option<Box<WasmResumeTrigger>>,
     pub update_layout: bool,
+    pub pre_run: Option<Box<TaskWasmPreRun>>,
```

Comment:
This is technically redundant , because one could just wrap the `run` and replace it, which would keep complexity down a bit.

BUT it also makes things more confusing , so this is probably better.

Reply:
That's a fair point... Initially, I wanted to make this fallible, but through multiple revisions ended up with this, which doesn't make a lot of sense.

Reply:
Although, I think the main callback isn't async?

Reply:
yep, the `run` field isn't async, and I don't want to make it async since it's used in many places. Let's just keep this.